ui/app/helpers/utils/util.js

```diff
@@ -275,6 +285,44 @@ export function shortenAddress (address = '') {
   return `${address.slice(0, 6)}...${address.slice(-4)}`
 }
 
+/**
+ * Shortens a URL for display, preserving the beginning and end.
+ * Returns the given URL if it is no longer than maxLength characters.
+ * Shortened URLs are maxLength + 3 characters long.
+ *
+ * Example output: llanfairpwllgw...ogogochuchaf.eu
+ *
+ * @param {string} url - The URL to shorten.
+ * @param {string} opts - Options bag.
+ * @param {number} opts.maxLength - The maximum length of the address.
+ * @param {boolean} opts.stripWwwPrefix - Whether to strip 'www.' from the address if present.
+ * @returns {string} The shortened URL, or the original if it was no longer
+ * than maxLength characters.
+ */
+export function shortenUrl (url = '', {
+  maxLength = 30,
+  stripWwwPrefix = true,
+} = {}) {
+  if (maxLength < 10) {
+    // it gets nonsensical somewhere around here
+    throw new Error('Expected maxLength >= 10')
+  }
+
+  let _url = url
+
+  if (stripWwwPrefix && _url.startsWith('www.')) {
```

Comment:
Note that we'd only want to strip `www` if it's a subdomain. We don't want to strip `www` from `www.eth` for example.

Reply:
I got rid of this entire function and handled it with CSS, instead.

Reply:
It probably would still be better to strip `www` subdomains when rendering the hostname for display, but it's not that important. Maybe we can revisit this when we move to using origins for site permissions instead of the hostname. Then we'll also want to hide default ports and schemes (e.g. https and 443), so we can take that opportunity to hide `www` as well.